src/components/select/SelectCommons.tsx

```diff
@@ -0,0 +1,329 @@
+/*
+ * Echoes React
+ * Copyright (C) 2023-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import styled from '@emotion/styled';
+import { Select as MantineSelect } from '@mantine/core';
+import { ComponentProps, forwardRef } from 'react';
+import { useIntl } from 'react-intl';
+import { isDefined } from '~common/helpers/types';
+import { PropsWithLabels } from '~types/utils';
+import { IconChevronDown, Spinner } from '..';
+import { INPUT_SIZE_VALUES, InputSize } from '../../utils/inputs';
+import { useSelectItemComponent } from './SelectItemCommons';
+import { SelectHighlight, SelectOption, SelectOptionType } from './SelectTypes';
+
+type MantineSelectProps = ComponentProps<typeof MantineSelect>;
+export interface SelectBaseProps {
+  className?: string;
+  data: ReadonlyArray<SelectOption>;
+  defaultValue?: MantineSelectProps['defaultValue'];
+  filter?: MantineSelectProps['filter'];
+  hasError?: boolean;
+  highlight?: SelectHighlight;
+  isDisabled?: boolean;
+  isLoading?: boolean;
+  isNotClearable?: boolean;
+  isSearchable?: boolean;
+  isRequired?: boolean;
+  labelError?: MantineSelectProps['error'];
+  labelNotFound?: MantineSelectProps['nothingFound'];
+  limit?: MantineSelectProps['limit']; // might change for a max height
```

Comment:
so which is it? limit or max height? We probably shouldn't leave this comment here